fabric-dimensions-v1/src/main/java/net/fabricmc/fabric/mixin/dimension/ServerBugfixMixin.java

```diff
@@ -0,0 +1,72 @@
+package net.fabricmc.fabric.mixin.dimension;
+
+import com.mojang.serialization.Lifecycle;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.ModifyVariable;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import net.minecraft.nbt.Tag;
+import net.minecraft.resource.DataPackSettings;
+import net.minecraft.server.Main;
+import net.minecraft.util.dynamic.RegistryOps;
+import net.minecraft.util.registry.DynamicRegistryManager;
+import net.minecraft.world.gen.GeneratorOptions;
+import net.minecraft.world.level.LevelInfo;
+import net.minecraft.world.level.LevelProperties;
+import net.minecraft.world.level.storage.LevelStorage;
+
+/**
+ * This Mixin aims to solve a Minecraft Vanilla bug where datapacks are ignored during creation of the
+ * initial LevelProperties when a dedicated server creates a completely new level.
+ *
+ * <p>This also includes the datapacks of loaded Fabric mods, and results in modded dimensions only
+ * being available after restarting the server, once the world has been created.
+ *
+ * <p>This Mixin aims to solve this problem by saving and loading the level.dat file once, after
+ * a new set of level properties is created. This will apply the same logic as reloading the
+ * level.dat after a restart, now including all datapack dimensions.
+ *
+ * <p>See https://bugs.mojang.com/browse/MC-195468 for a related bug report.
+ *
+ * <p>In 1.17: Retest if this bug still occurs without this Mixin by launching a dedicated server with the
+ * dimension testmod, and no world directory. If the dimension is available (i.e. in /execute in, or via
+ * the testmod's commands), then the bug is fixed and this Mixin can be removed.
+ */
+@Mixin(value = Main.class, remap = false)
+public class ServerBugfixMixin {
+	private static LevelStorage.Session fabric_session;
+
+	private static DynamicRegistryManager.Impl fabric_dynamicRegistry;
+
+	private static RegistryOps<Tag> fabric_registryOps;
```

Comment:
tiny nitpick but shouldn't these be `@Unique`?

Reply:
I don't know? Maybe? :-D
Doesn't @Unique just scramble the name and make the fabric_ prefix superflous? I thought adding the fabric_ prefix as per mod-id prefixing conventions was already sufficient.

Reply:
Agree with shartte. unique annotation is overrated.

Reply:
Why not just use @Unique and ignore the prefix?